.github/workflows/checks.yml

```diff
@@ -0,0 +1,63 @@
+name: Checks
+
+on:
+  push:
+    branches:
+      - 'main'
+  pull_request:
+    branches:
+      - 'main'
+  workflow_dispatch:
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.job }}-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  check-linting:
```

Comment:
Given how GH builds the display string, this'll wind up being "Checks / check-linting" - might as well drop the `check-` prefix? ie "Checks / linting" or just "Checks / lint"

This is up to you though